Applications/ApplicationsLib/UncoupledProcessesTimeLoop.h

```diff
@@ -0,0 +1,365 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef APPLICATIONSLIB_UNCOUPLED_PROCESSES_TIMElOOP
+#define APPLICATIONSLIB_UNCOUPLED_PROCESSES_TIMElOOP
+
+#include <memory>
+
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/ConfigTree.h"
+#include "NumLib/ODESolver/TimeDiscretizedODESystem.h"
+#include "NumLib/ODESolver/NonlinearSolver.h"
+
+#include "ProjectData.h"
+
+
+namespace ApplicationsLib
+{
+
+//! Time loop capable of time-integrating several uncoupled processes at once.
+template<typename Matrix, typename Vector>
+class UncoupledProcessesTimeLoop
+{
+public:
+    bool loop(ProjectData& project, std::string const& outdir);
+
+private:
+    //! An abstract nonlinear solver
+    using NLSolver       = NumLib::NonlinearSolverBase<Matrix, Vector>;
+    //! An abstract equations system
+    using EquationSystem = NumLib::EquationSystem;
+    //! An abstract process
+    using Process        = ProcessLib::Process<GlobalSetupType>;
+    //! An abstract time discretization
+    using TimeDisc       = NumLib::TimeDiscretization<Vector>;
+
+    struct PerProcessData
+    {
+        template<NumLib::ODESystemTag ODETag, NumLib::NonlinearSolverTag NLTag>
+        PerProcessData(NumLib::NonlinearSolver<Matrix, Vector, NLTag>& nl_slv,
+                       TimeDisc& time_disc,
+                       NumLib::ODESystem<Matrix, Vector, ODETag, NLTag>& ode_sys
+                       )
+            : nonlinear_solver_tag(NLTag)
+            , nonlinear_solver(nl_slv)
```

Comment:
I always align initializer lists like this and method argument lists at word boundaries. So it's consistent. ^^
I don't want to change my style here.